src/components/Loading.react.js

```diff
@@ -0,0 +1,148 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import GraphSpinner from '../fragments/Loading/spinners/GraphSpinner.jsx';
+import DefaultSpinner from '../fragments/Loading/spinners/DefaultSpinner.jsx';
+import CubeSpinner from '../fragments/Loading/spinners/CubeSpinner.jsx';
+import CircleSpinner from '../fragments/Loading/spinners/CircleSpinner.jsx';
+import DotSpinner from '../fragments/Loading/spinners/DotSpinner.jsx';
+import R from 'ramda';
+
+function getSpinner(type) {
+    switch (type) {
+        case 'graph':
+            return GraphSpinner;
+        case 'cube':
+            return CubeSpinner;
+        case 'circle':
+            return CircleSpinner;
+        case 'dot':
+            return DotSpinner;
+        default:
+            return DefaultSpinner;
+    }
+}
+
+/**
+ * A Loading component that wraps any other component and displays a spinner until the wrapped component has rendered.
+ */
+export default class Loading extends Component {
+    render() {
+        const {
+            loading_state,
+            color,
+            className,
+            style,
+            fullscreen,
+            debug,
+            type,
+        } = this.props;
+
+        const isLoading = getLoadingState(this);
+
+        if (isLoading) {
+            const Spinner = getSpinner(type);
+            return (
+                <Spinner
+                    className={className}
+                    style={style}
+                    status={loading_state}
+                    color={color}
+                    debug={debug}
+                    fullscreen={fullscreen}
+                />
+            );
+        }
+        if (
+            R.type(this.props.children) !== 'Object' ||
+            R.type(this.props.children) !== 'Function'
+        ) {
+            return <div className={className}>{this.props.children}</div>;
```

Comment:
We're not using the id passed through props for id+key of this component. Might cause similar problems as what we've seen with other comps

Reply:
I don't think we're setting `key` anywhere in the components, so I don't think it'll cause problems. The `key` is set in dash-renderer, however. But, you're right - I should add the `id` prop here!